src/routes/safe/components/Settings/ManageOwners/ReplaceOwnerModal/index.jsx

```diff
@@ -40,27 +41,38 @@ export const sendReplaceOwner = async (
   safeAddress: string,
   ownerAddressToRemove: string,
   ownerNameToRemove: string,
-  owners: List<Owner>,
+  ownersOld: List<Owner>,
   openSnackbar: Function,
   createTransaction: Function,
   updateSafe: Function,
 ) => {
   const gnosisSafe = await getGnosisSafeInstanceAt(safeAddress)
-  const storedOwners = await gnosisSafe.getOwners()
-  const index = storedOwners.findIndex(ownerAddress => ownerAddress === ownerAddressToRemove)
-  const prevAddress = index === 0 ? SENTINEL_ADDRESS : storedOwners[index - 1]
+  const safeOwners = await gnosisSafe.getOwners()
+  const index = safeOwners.findIndex(ownerAddress => ownerAddress.toLowerCase() === ownerAddressToRemove.toLowerCase())
+  const prevAddress = index === 0 ? SENTINEL_ADDRESS : safeOwners[index - 1]
   const txData = gnosisSafe.contract.methods
     .swapOwner(prevAddress, ownerAddressToRemove, values.ownerAddress)
     .encodeABI()
-  // const text = `Replace Owner ${ownerNameToRemove} (${ownerAddressToRemove}) with ${values.ownerName} (${values.ownerAddress})`
-
-  const ownerToRemoveIndex = owners.findIndex(o => o.address === ownerAddressToRemove)
-  const newOwner = makeOwner({ name: values.ownerName, address: values.ownerAddress })
 
   const txHash = await createTransaction(safeAddress, safeAddress, 0, txData, openSnackbar)
 
+  let owners = []
+  const storedOwners = await getOwners(safeAddress)
```

Comment:
why do we need this?

Reply:
This was added on all the operations (remove, replace and add) to solve the issue discovered by Tobias some messages above in this PR. See option (2) in the message.

Reply:
1. ` ownersOld: List<Owner>,` is unused
2. If you want to get safe owners before updating the safe after tx is mined, this function should be rewritten as a redux action and there you can use a selector to get the safe from redux store. LocalStorage should be modified only via redux middleware so it's more predictable

Reply:
or you can make specific actions like `addOwner, replaceOwner, removeOwner` which won't use the owners list at all. You would just call `addOwner(safeAddress, owner)` and redux store will handle everything with the latest data. This solution feels quite hacky to me